internal/pagination/purge/purge.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package purge
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// RegisterJobs registers the purge job for each deletion table with the provided scheduler.
+func RegisterJobs(ctx context.Context, s *scheduler.Scheduler, r db.Reader, w db.Writer) error {
+	const op = "purge.RegisterJobs"
+	if s == nil {
+		return errors.New(ctx, errors.InvalidParameter, "nil scheduler", op, errors.WithoutEvent())
+	}
+	if util.IsNil(r) {
+		return errors.New(ctx, errors.Internal, "nil DB reader", op, errors.WithoutEvent())
+	}
+	if util.IsNil(w) {
+		return errors.New(ctx, errors.Internal, "nil DB writer", op, errors.WithoutEvent())
+	}
+
+	rows, err := r.Query(ctx, "select get_deletion_tables()", nil)
```

Comment:
I considered it but I also wasn't sure if it was going to cause more confusion since it's a subpackage of the `pagination` package that is introduced in this PR. It seems that PR is mostly done and I don't want to add that code now since it wasn't reviewed by Jim and Mike when they approved the PR.